blocks/news-feed/news-feed.js

```diff
@@ -0,0 +1,104 @@
+import ffetch from '../../scripts/ffetch.js';
+import { createOptimizedPicture, decorateIcons } from '../../scripts/lib-franklin.js';
+
+function buildPost(post, eager) {
+  const postCard = document.createElement('div');
+  postCard.classList.add('news-card');
+
+  const description = post.content.querySelector('h1 + p');
+  const title = post.content.querySelector('h1').textContent;
+
+  postCard.innerHTML = `
+      <div class="news-image">
+        <a href="${post.path}">${createOptimizedPicture(post.image, `Teaser image for ${title}`, eager).outerHTML}</a>
+      </div>
+      <div class="news-content">
+        <a class="post-title" href="${post.path}">${title}</a>
+        <a class="post-description" href="${post.path}">
+          <p>${description ? description.textContent : post.description}</p>
+          <span>Read More</span>
+        </a>
+      </div>
+    </a>
+  `;
+
+  return postCard;
+}
+
+async function buildNewsFeed(ul, pageNum, pageControl) {
+  const limit = 5;
+  const offset = pageNum * limit;
+  let morePages = false;
+  const blogPosts = ffetch('/query-index.json')
```

Comment:
I'd rename `query-index` to something like `news-feed.json` and also leave a comment in the code that its content is maintained in `news-feed.xlsx` - whoever ends up maintaining this site on RiteAid and doesn't know Hlx that well, to find it more manageable.

Reply:
Also if the spreadsheet is to be maintained by authors, can we change the columns `lastModified` and `date` into a Date cell format in excel ?

Reply:
I'd also rename the var `blogPosts` into `newsArticles`

Reply:
`query-index` is a standard name across all projects, so I'm disinclines to rename that. the content is not author maintined, see https://www.hlx.live/developer/indexing


renaming the variable is a good idea, will make that change